### PR TITLE
HTS-403 : Encrypt email before saving to mongodb

### PR DESCRIPTION
**Pull request by sureshhmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/helptosavefrontend/repo/EnrolmentStore.scala [hunk 1]
```diff
@@ -70,34 +73,59 @@ class MongoEnrolmentStore @Inject()(mongo: ReactiveMongoComponent)(implicit ec:
...
 
-  private[enrolment] def doUpdate(data: EnrolmentData)(implicit ec: ExecutionContext): Future[Option[EnrolmentData]] =
+  private[repo] def doCreate(nino: NINO, itmpFlag: Boolean, email: String)(implicit ec: ExecutionContext): Future[Option[EnrolmentData]] =
     collection.findAndUpdate(
-      BSONDocument("nino" -> data.nino),
-      BSONDocument("$set" -> BSONDocument("itmpHtSFlag" -> data.itmpHtSFlag)),
+      BSONDocument("nino" -> nino),
+      BSONDocument("$set" -> BSONDocument("itmpHtSFlag" -> itmpFlag, "email" -> encrypt(email))),
+      fetchNewObject = true,
+      upsert = true
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosavefrontend/repo/EnrolmentStore.scala]: do we want to `upsert` in this case? We're creating, we shouldn't be updating existing entries

File: app/uk/gov/hmrc/helptosavefrontend/repo/EnrolmentStore.scala [hunk 2]
```diff
@@ -78,7 +78,7 @@ class MongoEnrolmentStore @Inject()(mongo: ReactiveMongoComponent)(implicit ec:
       BSONDocument("nino" -> nino),
       BSONDocument("$set" -> BSONDocument("itmpHtSFlag" -> itmpFlag, "email" -> encrypt(email))),
       fetchNewObject = true,
-      upsert = true
+      upsert = false
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/repo/EnrolmentStore.scala]: please hold

Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosavefrontend/repo/EnrolmentStore.scala]: whoops, sorry missed the fact it said `false` there

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**